include/d/actor/d_a_obj_swpush.h

```diff
@@ -1,37 +1,105 @@
 #ifndef D_A_OBJ_SWPUSH_H
 #define D_A_OBJ_SWPUSH_H
 
-#include "f_op/f_op_actor.h"
+#include "d/d_a_obj.h"
+#include "m_Do/m_Do_ext.h"
 #include "d/d_bg_w.h"
 
+class dBgWSv;
 class J3DNode;
 
 namespace daObjSwpush {
+    enum AttrFlag_e {
+        FLAG_OBEY_SAVE = 0x1,
+        FLAG_STAY_PRESSED = 0x2,
+        FLAG_IS_TOGGLE = 0x4,
+        FLAG_REQ_HEAVY = 0x8,
+        FLAG_ON_IS_UP = 0x10,
+        FLAG_UNK20 = 0x20,
+    };
+
+    struct Attr_c {
+        /* 0x00 */ u32 mHeapSize;
+        /* 0x04 */ AttrFlag_e mFlags;
+        /* 0x08 */ float mScale;
+        /* 0x0C */ const char* mKbotaResName;
+        /* 0x10 */ const char* mHhbotResName;
+        /* 0x14 */ const char* mBgArcName;
+        /* 0x18 */ const char* mModelArcName;
+        /* 0x1C */ const char* mBtpArcName;
+        /* 0x20 */ short mBgResIndex;
+        /* 0x22 */ short mModelResIndices[2];
+        /* 0x26 */ short mBtpResIndex;
+        /* 0x28 */ float mSpring;
+        /* 0x2C */ float mSpeedDecay;
+        /* 0x30 */ float mPushSpeed0;
+        /* 0x34 */ float m34;
+        /* 0x38 */ short m38;
+        /* 0x3A */ short mMiniPushDelay1;
+        /* 0x3C */ short mPushDelay;
+        /* 0x3E */ short mMiniPushDelay2;
+        /* 0x40 */ float m40;
+        /* 0x44 */ float m44;
+        /* 0x48 */ short mPauseDuration;
```

Comment:
We should probably use the `s16`/`f32`/etc dolphin typedefs instead of the full type names for primitives. Assertion strings throughout the game indicate the devs used those pretty consistently

Reply:
does `int` have an equivalent typedef like that? I tried changing an `int` parameter to `s32` (specifically the second argument to jnodeCB) and got an error:
`Cannot initialize a parameter of type 'J3DNodeCallBack' (aka 'int (*)(J3DNode *, int)') with an lvalue of type 'BOOL (J3DNode *, s32)' (aka 'int (J3DNode *, long)'): type mismatch at 2nd parameter ('int' vs 's32' (aka 'long'))`

Reply:
No, don't convert `int` to `s32`, leave it as is. Even though `int` and `s32` are both signed 32 bit integers, the compiler treats them differently in a bunch of ways (not just errors, it sometimes produces different code that doesn't match if you use the wrong one).